runtime/src/append_vec.rs

```diff
@@ -899,16 +899,16 @@ pub mod tests {
         let accounts = av.accounts(0);
         let account = accounts.first().unwrap();
 
+        // upper 7-bits are not 0, so sanitization should fail
+        assert!(!account.sanitize_executable());
+
         // we can observe crafted value by ref
         {
             let executable_bool: &bool = &account.account_meta.executable;
-            // Depending on use, *executable_bool can be truthy or falsy due to direct memory manipulation
-            // assert_eq! thinks *executable_bool is equal to false but the if condition thinks it's not, contradictorily.
+            // Depending on use, *executable_bool can be truthy or falsy due to direct memory manipulation.
+            // Behavior is dependent on compiler.
+            // assert! thinks !*executable_bool is true, for all compilers
             assert!(!*executable_bool);
-            const FALSE: bool = false; // keep clippy happy
-            if *executable_bool == FALSE {
-                panic!("This didn't occur if this test passed.");
-            }
```

Comment:
could you restore this with `#[cfg(!m1)]`?

the intent isn't clear but this test is checking the existence of undefined behavior around bool inside rustc (or llvm): https://doc.rust-lang.org/reference/types/boolean.html:

> An object with the boolean type has a [size and alignment](https://doc.rust-lang.org/reference/type-layout.html#size-and-alignment) of 1 each. The value false has the bit pattern 0x00 and the value true has the bit pattern 0x01. It is [undefined behavior](https://doc.rust-lang.org/reference/behavior-considered-undefined.html) for an object with the boolean type to have any other bit pattern.

as you know, we're not relying on the UB (bit pattern other than 0x00 and 0x01 for `bool`) in any production code. however, this assertion is needed to validate that this test is still exposing the UB to make sure we're not relaying on it accidentally. So, if possible, i want to the test to check the existence of UB at least for x64.

any ad-hoc `cfg` should be fine as long as our CI is happy.

Reply:
Gotcha, re-added the if/panic, but disabled it for aarch64.